src/classic/element/ReactElement.js

```diff
@@ -228,6 +228,60 @@ ReactElement.cloneAndReplaceProps = function(oldElement, newProps) {
   return newElement;
 };
 
+ReactElement.cloneElement = function(element, config, children) {
+  var propName;
+
+  // Original props are copied
+  var props = assign({}, element.props);
+
+  // Reserved names are extracted
+  var key = element.key;
+  var ref = element.ref;
+
+  // Owner will be preserved, unless ref is overridden
+  var owner = element._owner;
+
+  if (config != null) {
+    if (config.ref !== undefined) {
+      // Silently steal the ref from the parent.
+      ref = config.ref;
+      owner = ReactCurrentOwner.current;
+    }
+    if (config.key !== undefined) {
```

Comment:
hm... Nah. Maybe next one. I think that this is still consistent with key/ref having a default value of `null`.

Imagine these ES6 signatures:

```
createElement(type, { ref = null, key = null, ...props }, ...children);
```
```
cloneElement(element, { ref = element.ref, key = element.key, ...props }, ...children);
```